wear/src/main/java/io/homeassistant/companion/android/phone/PhoneSettingsListener.kt

```diff
@@ -216,7 +216,8 @@ class PhoneSettingsListener : WearableListenerService(), DataClient.OnDataChange
     private fun saveTileTemplate(dataMap: DataMap) = mainScope.launch {
         val content = dataMap.getString(WearDataMessages.CONFIG_TEMPLATE_TILE, "")
         val interval = dataMap.getInt(WearDataMessages.CONFIG_TEMPLATE_TILE_REFRESH_INTERVAL, 0)
-        wearPrefsRepository.setTemplateTileContent(content)
+        // TODO: combine to one call
+        wearPrefsRepository.setTemplateTile(content)

```

Comment:
I've read the documentation on https://developer.android.com/training/wearables/data/events and on other pages in the `Handling data` section, but I'm not still not sure exactly when/why this method is called :thinking: It's called from `onDataChanged` in this class, but when is that method called?

Reply:
Ah, I somehow assumed the `PhoneSettingsListener` class was run on the companion device, but it's in fact run on the wearable device. So this method is called whenever the settings have been changed on the companion device, right?